contracts/BytesLib.sol

```diff
@@ -286,6 +286,39 @@ library BytesLib {
         return tempAddress;
     }
 
+    function toUint8(bytes _bytes, uint _start) internal  pure returns (uint8) {
+        require(_bytes.length >= (_start + 1));
+        uint8 tempUint;
+
+        assembly {
+            tempUint := mload(add(add(_bytes, 0x1), _start))
```

Comment:
I can see how this might be confusing looking at all the other examples but the 32 (`0x20`) bytes being added to `_bytes` has nothing to do with the size of the casting target type.
As you might notice in the typecast to `address` method, we also add `0x20` to `_bytes`.

We do the above to bypass the first 32 bytes of the bytes array. Why? Because it's how dynamic arrays keep track of their own length in memory. The first 32 bytes actually are a _uint256_ that holds the array's length.

So basically here we should do the same and add `0x20`.

Why are the tests passing then? Because this addition is effectively making us compare the most significant bytes of a `uint256` that holds the array's length as we discussed above. Effectively making us compare `0x00..00` to `0x00..00`.